src/main/java/com/google/codeu/data/Message.java

```diff
@@ -25,20 +25,23 @@
   private String user;
   private String text;
   private long timestamp;
+  private String recipient;
 
   /**
-   * Constructs a new {@link Message} posted by {@code user} with {@code text} content. Generates a
-   * random ID and uses the current system time for the creation time.
+   * Constructs a new {@link Message} posted by {@code user} with {@code text} content 
+   * and recipient {@code recipient}. Generates a random ID and uses the current 
+   * system time for the creation time.
    */
-  public Message(String user, String text) {
-    this(UUID.randomUUID(), user, text, System.currentTimeMillis());
+  public Message(String user, String text, String recipient) {
+    this(UUID.randomUUID(), user, text, System.currentTimeMillis(), recipient);
   }
 
-  public Message(UUID id, String user, String text, long timestamp) {
+  public Message(UUID id, String user, String text, long timestamp, String recipient) {
```

Comment:
Sigh. This class. At some point we should probably make all of the data members final, change the constructor to be private, and use a builder pattern.  I'll see if I can dig up a presentation I had about that for next meeting.  (I'm not requesting changes here, just giving you a heads-up that I'll either go in and do this or find out if anybody else wants to.)